### PR TITLE
Added numpy linear algebra einsum method.

### LINKED ISSUE #18930: einsum

### PR DESCRIPTION
**Pull request by Ashwani132003**
Close #18930

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by MuhammedAshraf2020** [comment]
Hi @Ashwani132003,
Thanks for the PR. There seems to be a few small issues that I pointed out below.

File: ivy/functional/frontends/numpy/linalg/matrix_and_vector_products.py [hunk 1]
```diff
@@ -75,3 +75,10 @@ def multi_dot(arrays, *, out=None):
 def dot(a, b, out=None):
     a, b = promote_types_of_numpy_inputs(a, b)
     return ivy.matmul(a, b)
+
+@handle_numpy_out
+@to_ivy_arrays_and_back
+def einsum(
```
Comment (MuhammedAshraf2020) [inline ivy/functional/frontends/numpy/linalg/matrix_and_vector_products.py:81]: In the frontend functions, it is essential to ensure that the function name and call signature precisely match the corresponding function in the target framework. The original function from the target framework is 
`def einsum(*operands, out=None, optimize=False, **kwargs)` [source](https://github.com/numpy/numpy/blob/v1.25.0/numpy/core/einsumfunc.py#L1010). Please refer to the frontend documentation [here](https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends.html) for assistance. However, it appears that your call signature differs from the original function.

**Review by MuhammedAshraf2020 — CHANGES_REQUESTED** [review]

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**